### DISCUSSION TITLE
Why quine?

### DISCUSSION THREAD

**Discussion by jan-glx** [yachielab/QUEEN#12] [General] [answered]
I was wondering what the value of the quinable feature of dna-queen is? 
Isn't the original code always more valuable than the code automatically produced based on some history fields, since it may have comments and variable names? 
Wouldn't dropping support of the quineabilty feature greatly simplify the implementation and perhaps even the interface? 
I understand that the quinability was considered a core feature of dna-queen but I think ultimately a researcher is much less interested in how a plasmid was generated and much more in how they can generate that plasmid -  something very different and that dna-queen is very useful without it...

**Comment by ponnhide** [comment, marked as answer]
>  but I think ultimately a researcher is much less interested in how a plasmid was generated and much more in how they can generate that plasmid

You're right (Actually, I sometimes take similllar comments from others.. haha.). Some DNA design tools, including QUEEN (such as Snapgene and Benchling), also implement the function to track and reference the construction history of DNA constructs. These details are considered valuable to ensure the reusability, safety, and copyright of DNA sequences. However, many users are primarily concerned with whether the DNA construct functions, rather than its detailed genetic attributes and construction history. The importance of this information, such as copyrights and licenses that should ideally accompany DNA constructs, will probably be recognized after more discussion about this topic in the academic community.

On the other hand, I believe the quine function of QUEEN is also crucial in the context of laboratory automation. Automating DNA synthesis experiments requires automating not just the experimental procedures but also the design processes of DNA sequences and construction protocols. Currently, many researchers spend a lot of time not only on experiments but also on the design processes. As a solution to solve this issue, we can suggest developing AI tools that can learn these design processes. However, there is a lack of adequate and usable data that clearly links DNA constructs and their design processes. Thus, while it may still seem like a far-future idea, by promoting the use of QUEEN, I aim to accumulate learning data on design processes of DNA sequences and their construction protocols within the research community.

I acknowledge that the quine function of QUEEN certainly complicates the implementation, but the design process data obtained through this function will be invaluable for the development of future AI tools. Despite feeling the limits of personal development, I believe the "quine" idea is a key to realizing the purpose.


By the way, as you pointed out, I had added non-proper modification into the flipdna function due to my confusion for my implementation...  
For now, this issue was re-modified. Thank you for pointing this out.
